lib/matplotlib/text.py

```diff
@@ -1986,107 +1986,115 @@ class Annotation(Text, _AnnotationBase):
     def __str__(self):
         return "Annotation(%g, %g, %r)" % (self.xy[0], self.xy[1], self._text)
 
-    @docstring.dedent_interpd
     def __init__(self, s, xy,
                  xytext=None,
                  xycoords='data',
                  textcoords=None,
                  arrowprops=None,
                  annotation_clip=None,
                  **kwargs):
-        '''
-        Annotate the point ``xy`` with text ``s``.
+        """
+        Annotate the point *xy* with text *s*.
+
+        In the simplest form, the text is placed at *xy*.
 
-        Additional kwargs are passed to `~matplotlib.text.Text`.
+        Optionally, the text can be displayed in another position *xytext*.
+        An arrow pointing from the text to the annotated point *xy* can then
+        be added by defining *arrowprops*.
 
         Parameters
         ----------
-
         s : str
             The text of the annotation.
 
-        xy : iterable
-            Length 2 sequence specifying the *(x,y)* point to annotate.
+        xy : 2-tuple of float
```

Comment:
I don't think it needs to be a 2-tuple, and neither a float, e.g. the following are all valid

```
a = mtext.Annotation("String", xy=[.5,.2], xytext=np.array([.65,.32]))
b = mtext.Annotation("String", xy=range(2), xytext={3,5})
```

Reply:
I'm aware of this.

#### float

We've decided to
> Use float for a type that can be any number.

https://matplotlib.org/devel/documenting_mpl.html#parameter-type-descriptions

https://github.com/matplotlib/matplotlib/issues/10225#issuecomment-360975571

#### tuple

While iterable is technically correct, it lacks all the relevant information (required fixed size of 2 and type). Technically, you would have to write `iterable of float of length 2` or `length-2 iterable of float`, which is IMO is not quite readable and scary to less experienced programmers. Lot's of other docstrings use tuple as well even though you can pass other iterables. I'm willing to sacrifice precision of the definition for simplicity here.

We do not have a consistent and good way of describing this (should be added to the [documentation recommendations](https://matplotlib.org/devel/documenting_mpl.html#parameter-type-descriptions) once we've reached a decision). Some other options used in the code base (modulo float <-> scalar) are:

- `tuple of 2 floats`
- `(float, float)`
- `tuple of (float, float)`

Thinking again, maybe `(float, float)` is simple and clear enough. It concisely defines contained type and size, and does not explicitly mention the word tuple.